src/DDTrackCreatorBase.cc

```diff
@@ -421,16 +421,34 @@ void DDTrackCreatorBase::GetTrackStatesAtCalo( EVENT::Track *track,
   auto marlintrk = std::unique_ptr<MarlinTrk::IMarlinTrack>(m_trackingSystem->createTrack());
   const EVENT::TrackerHitVec& trkHits = track->getTrackerHits();
   const int nHitsTrack = trkHits.size();
+
   for (int iHit = 0; iHit < nHitsTrack; ++iHit) {
-      marlintrk->addHit(trkHits[iHit]);
+    EVENT::TrackerHit* trkHit = trkHits[iHit] ;
+    if( UTIL::BitSet32( trkHit->getType() )[ UTIL::ILDTrkHitTypeBit::COMPOSITE_SPACEPOINT ]   ){ //it is a composite spacepoint
+      //Split it up and add both hits to the MarlinTrk
+      const EVENT::LCObjectVec rawObjects = trkHit->getRawHits();
+      for( unsigned k=0; k< rawObjects.size(); k++ ){
+	EVENT::TrackerHit* rawHit = dynamic_cast< EVENT::TrackerHit* >( rawObjects[k] );
```

Comment:
`static_cast` or check `rawHit` for `nullptr`, or coverity will complain.

Reply:
done.